accounting/invoice/choices.py

```diff
@@ -56,3 +60,126 @@ def approved_choices(cls):
+        """
+        Return the default value given to the `default` choice.
+        """
+        return 'OC-{}'.format(cls.default_number())
+
+    @classmethod
+    def increment_default(cls, value):
+        """
+        Increment the number for the 'default' numbering scheme.
+
+        Assumes `value` is in the format of the 'default' numbering scheme.
+        """
+        # Turn the input value string into a datetime object.
+        datetime_obj = datetime.datetime.strptime(value, cls.default)
+        # Increment it one month forward.
+        incremented_obj = datetime_obj + datetime.timedelta(days=32)
+        # Turn the object back into a string using the numbering scheme.
+        return incremented_obj.strftime(cls.default)
+
+    @classmethod
+    def increment_number(cls, value):
+        """
+        Increment the number for the 'number' numbering scheme.
+
+        Assumes `value` is in the format of the 'number' numbering scheme.
+        """
+        return cls.number.format(number=int(value) + 1)
+
+    @classmethod
+    def increment_year_month_number(cls, value):
+        """
+        Increment the number for the 'year_month_number' numbering scheme.
+
+        Assumes `value` is in the format of the 'year_month_number' numbering scheme.
+        """
+        date, number = value.rsplit('-', 1)
+        incremented_date = cls.increment_default(date)
+        incremented_number = cls.increment_number(number)
+        return '-'.join([incremented_date, incremented_number])
+
+    @classmethod
+    def increment_opencraft_year_month(cls, value):
+        """
+        Increment the number for the 'opencraft_year_month' numbering scheme.
+
+        Assumes `value` is in the format of the 'opencraft_year_month' numbering scheme.
+        """
+        prefix, date = value.split('-', 1)
+        incremented_date = cls.increment_default(date)
+        return '-'.join([prefix, incremented_date])
+
+    @classmethod
+    def increment_opencraft_number(cls, value):
+        """
+        Increment the number for the 'opencraft_number' numbering scheme.
+
+        Assumes `value` is in the format of the 'opencraft_number' numbering scheme.
+        """
+        prefix, number = value.split('-', 1)
+        incremented_number = cls.increment_number(number)
+        return '-'.join([prefix, incremented_number])
+
+    @classmethod
+    def increment_value(cls, choice, value=None):
+        """
+        Increment any passed in value given its matching numbering scheme choice.
+
+        Assumes, if given:
+        * `choice` is one of the value strings associated with a choice, not the `ChoiceItem` object
+        * `value` is in the format of the choice's numbering scheme.
+        """
+        function_name = cls.attributes[choice]
+        function = getattr(InvoiceNumberingScheme, 'increment_{}'.format(function_name))
+        if value is None:
+            value = getattr(InvoiceNumberingScheme, 'default_{}'.format(function_name))()
```

Comment:
@UmanShahzad Under what conditions will `value` be `None`? It could be worth mentioning that here. If it'll be `None` if and only if we're generating an invoice for a team member for the first time, shouldn't this method return the default value? The way it's currently written it seems like it would increment the default value before returning it.

Reply:
@itsjeyd Oh damn, yeah, I mixed up the reason I even implemented it this way: in the `send_email_with_invoice` function, when we're generating for the first time, we can pass in `None` to some `default_value` function, not an *increment* function. I will change this and its use case.